Replace HttpUtility.UrlEncode with Uri.EscapeDataString, use escaped form of + when joining multiple structured logs filters

adamint opened the pull request:
`Uri.EscapeDataString` is more appropriate than `HttpUtility.UrlEncode` in our uses, as we are not using it to encode paths. 

This fixes #2762 as well, as a space is encoded by UrlEncode as a +, instead of %20. Since the separator between filters was +, multi-word filters were not working.

<img width="1423" alt="image" src="https://github.com/dotnet/aspire/assets/20359921/49df7f90-acd9-46c7-ae11-c5265d795e57">

The filters query parameter for the above now encodes to `filters=Message:contains:Application%20started%2BApplication:!contains:f`

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/2817)

adamint committed:
Replace HttpUtility.UrlEncode with Uri.EscapeDataString, use escaped form of + when building compound filters

JamesNK reviewed (changes requested):
Add some tests to verify the behavior you want from using `Uri.EscapeDataString`.

Commit:
Update dashboard url tests to test multiple filters in logs

adamint commented:
> Add some tests to verify the behavior you want from using `Uri.EscapeDataString`.

Added a test, as well as updated the existing structured log test to use `LogFilterFormatter.SerializeLogFiltersToString`. I also added a comment to explain why there are some encoding differences with the same input string.

JamesNK reviewed (approved):
Looks good. Also add a test for spaces in values before merging: https://github.com/dotnet/aspire/issues/2846

Commit:
Add filter value with spaces structured url test
